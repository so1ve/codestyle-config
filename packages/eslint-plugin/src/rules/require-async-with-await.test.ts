import { run } from "./_test";
import rule, { RULE_NAME } from "./require-async-with-await";

const valid = [
	"async () => { await 1; }",
	"async function a() { await 1; }",
	"class a { async a() { await 1; } }",
	"class a { static async a() { await 1; } }",
	"class a { constructor() { await 1; } }", // We don't handle constructor
	"class a { get a() { await 1; }; set a() { await 1; } }", // nor getters and setters
	"const a = { get a() { await 1; }, set a() { await 1; } }", // and inside an object
];

const invalid = [
	"() => { await 1; }",
	"function a() { await 1; }",
	"() => { function a() { await 1; } }",
	"() => { () => {}; await 1; }",
	"class a { a() { await 1; } }",
	"class a { @foo a() { await 1; } }",
	"class a { static a() { await 1; } }",
	"class a { @foo static a() { await 1; } }",
	"class a { @foo static a() { await 1; } }",
	"class a { a = function a() { await 1; } }",
	"const a = { a() { await 1; } }",
];

run({
	name: RULE_NAME,
	rule,
	valid,
	invalid,
});
