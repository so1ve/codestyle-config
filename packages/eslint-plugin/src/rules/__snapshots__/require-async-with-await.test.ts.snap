// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`require-async-with-await > invalid > () => { () => {}; await 1; } 1`] = `"async () => { () => {}; await 1; }"`;

exports[`require-async-with-await > invalid > () => { await 1; } 1`] = `"async () => { await 1; }"`;

exports[`require-async-with-await > invalid > () => { function a() { await 1; } } 1`] = `"() => { async function a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { @foo a() { await 1; } } 1`] = `"class a { @foo async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { @foo static a() { await 1; } } 1`] = `"class a { @foo static async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { @foo static a() { await 1; } } 2`] = `"class a { @foo static async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { a = function a() { await 1; } } 1`] = `"class a { a = async function a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { a() { await 1; } } 1`] = `"class a { async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > class a { static a() { await 1; } } 1`] = `"class a { static async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > const a = { a() { await 1; } } 1`] = `"const a = { async a() { await 1; } }"`;

exports[`require-async-with-await > invalid > function a() { await 1; } 1`] = `"async function a() { await 1; }"`;
